### DISCUSSION TITLE
Enum underlying type any of struct

### DISCUSSION THREAD

**Discussion by Ekl1p5e** [dotnet/csharplang#1048] [General]
Currently, there is no way of having a enumerable constants that are structs or floats.

It would be useful to define an enum with an underlying type of simple or complex types in order to assign standard or common complex values to fields.

Example:
enum Window : Rectangle
{
 MinSize = { 400, 300, 0, 0 },
 MaxSize = {600, 500, 0, 0 }
}

**Comment by tannergooding** [comment]
This would require, at the very least, implementing the `blittable constant` feature. Not to mention changes to the runtime to support additional enum base types.

**Comment by orthoxerox** [comment]
Why should they be enums and not constants?

**Comment by Joe4evr** [comment]
Or simply `static readonly` members?

**Comment by MkazemAkhgary** [comment]
enums be enums, but constant arrays are missing feature, that would also open gate to have constant arrays of structs that can be constant.

**Comment by tannergooding** [comment]
@MkazemAkhgary, see https://github.com/dotnet/csharplang/issues/688

**Comment by Thaina** [comment]
realted : #144
